src/modules/map/components/openlayers/OpenLayersInternalLayer.vue

```diff
@@ -1,3 +1,45 @@
+<script setup>
+/**
+ * Transforms a layer config (metadata, structures can be found in api/layers/** files) into the
+ * correct OpenLayers counterpart depending on the layer type.
+ */
+
+import AbstractLayer from '@/api/layers/AbstractLayer.class'
+import LayerTypes from '@/api/layers/LayerTypes.enum'
+import OpenLayersExternalWMTSLayer from '@/modules/map/components/openlayers/OpenLayersExternalWMTSLayer.vue'
+import OpenLayersKMLLayer from '@/modules/map/components/openlayers/OpenLayersKMLLayer.vue'
+import { WEBMERCATOR } from '@/utils/coordinates/coordinateSystems'
+import { computed, toRef } from 'vue'
+import { useStore } from 'vuex'
+import OpenLayersGeoJSONLayer from './OpenLayersGeoJSONLayer.vue'
+import OpenLayersVectorLayer from './OpenLayersVectorLayer.vue'
+import OpenLayersWMSLayer from './OpenLayersWMSLayer.vue'
+import OpenLayersWMTSLayer from './OpenLayersWMTSLayer.vue'
+
+const props = defineProps({
+    layerConfig: {
+        type: AbstractLayer,
+        default: null,
+    },
+    zIndex: {
+        type: Number,
+        default: -1,
+    },
+})
+const layerConfig = toRef(props, 'layerConfig')
+const zIndex = toRef(props, 'zIndex')
```

Comment:
Would not be more concise to write
```javascript
const {layerConfig, zIndex} = toRefs(defineProps({
    layerConfig: {
        type: AbstractLayer,
        default: null,
    },
    zIndex: {
        type: Number,
        default: -1,
    },
}))
```

At least this would avoid copy paste mistake where the `toRef` would be forgotten. Changing everywhere now might be overkiling, especially where you split it into several statement with props rename, but here and adding this as example in the readme (in the existing example) might be a very good idea.